### PR TITLE
Geophile integration - punchlist items 1, 4, 5, 6, 7

### PR DESCRIPTION
**Pull request by geophile**
This depends on geophile 2.0.5, which is working its way through search.maven.org.

### CODE REVIEW

**Comment by nathanlws** [comment]
Does this supersede #1668 ?

File: fdb-sql-layer-core/src/main/java/com/foundationdb/qp/storeadapter/indexcursor/IndexCursorSpatial_InBox.java [hunk 1]
```diff
@@ -268,23 +276,34 @@ private void logQueryIndex(SortedArray<RecordWithSpatialObject> queryIndex)
...
+    public static SpatialJoin.InputObserver SPATIAL_JOIN_LEFT_OBSERVER = null;
+    public static SpatialJoin.InputObserver SPATIAL_JOIN_RIGHT_OBSERVER = null;
+    private static SpatialJoin.InputObserver SPATIAL_JOIN_RIGHT_DEBUG_OBSERVER =
         new SpatialJoin.InputObserver()
         {
             @Override public void randomAccess(Cursor cursor, long z)
             {
-                LOG.debug("Random access using {}: {}", cursor, SpaceImpl.formatZ(z));
+                // LOG.debug("Random access using {}: {}", cursor, SpaceImpl.formatZ(z));
+                System.out.format("Random access using %s: %s\n", cursor, SpaceImpl.formatZ(z));
```
Comment (nathanlws) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/qp/storeadapter/indexcursor/IndexCursorSpatial_InBox.java]: Does LOG.debug not work?

File: fdb-sql-layer-core/src/main/java/com/foundationdb/qp/storeadapter/indexcursor/IndexCursorSpatial_InBox.java [hunk 2]
```diff
@@ -268,23 +276,34 @@ private void logQueryIndex(SortedArray<RecordWithSpatialObject> queryIndex)
...
             public void sequentialAccess(Cursor cursor, long zRandomAccess, Record record)
             {
+/*
                 LOG.debug("    Sequential access using {} {} -> {}: {}",
                           cursor,
                           SpaceImpl.formatZ(zRandomAccess),
                           record == null ? SpaceImpl.Z_NULL : SpaceImpl.formatZ(record.z()),
                           record);
+*/
+                System.out.format("    Sequential access using %s %s -> %s: %s\n",
```
Comment (nathanlws) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/qp/storeadapter/indexcursor/IndexCursorSpatial_InBox.java]: LOG.debug would be preferable.

File: fdb-sql-layer-core/src/main/java/com/foundationdb/server/spatial/Spatial.java [hunk 3]
```diff
@@ -101,6 +103,11 @@ public static SpatialObject deserializeWKT(Space space, String string) throws Pa
             : JTS.spatialObject(space, geometry);
     }
 
+    static
+    {
+        // java.util.logging.Logger.getLogger("").setLevel(Level.FINE);
```
Comment (nathanlws) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/server/spatial/Spatial.java]: Debug attempt? Can probably just remove the whole static block if it isn't needed.

File: fdb-sql-layer-core/src/test/java/com/foundationdb/server/test/it/qp/SpatialQueryDT.java [hunk 4]
```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (C) 2009-2013 FoundationDB, LLC
```
Comment (nathanlws) [inline fdb-sql-layer-core/src/test/java/com/foundationdb/server/test/it/qp/SpatialQueryDT.java]: 2015

File: fdb-sql-layer-core/src/test/java/com/foundationdb/server/test/it/qp/SpatialQueryPerformanceDT.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (C) 2009-2013 FoundationDB, LLC
```
Comment (nathanlws) [inline fdb-sql-layer-core/src/test/java/com/foundationdb/server/test/it/qp/SpatialQueryPerformanceDT.java]: 2015

File: pom.xml [hunk 6]
```diff
@@ -109,7 +109,6 @@
                     </systemPropertyVariables>
                     <argLine>-Xmx512m ${fdbsql.test.extraJvmArgs} ${fdbsql.test.JmxJvmArgs} ${surefireJacocoArg}
                         -Dmvn.main.basedir=${main.basedir} -Dmvn.settings.localRepository=${settings.localRepository}
-                        -Djava.security.manager -Djava.security.policy=${main.basedir}/fdb-sql-layer-common/test/resources/tests.policy
```
Comment (nathanlws) [inline pom.xml]: Edits to the pom should be undone.

**Comment by nathanlws** [comment]
Also covers punchlist item 1, right?

Few small notes but a quick skim looks as described.

File: fdb-sql-layer-core/src/main/java/com/foundationdb/qp/storeadapter/indexcursor/IndexCursorSpatial_InBox.java (hunk 1 above)
Comment (geophile) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/qp/storeadapter/indexcursor/IndexCursorSpatial_InBox.java]: It does, but was inconvenient. I'll restore this.

File: fdb-sql-layer-core/src/main/java/com/foundationdb/qp/storeadapter/indexcursor/IndexCursorSpatial_InBox.java (hunk 2 above)
Comment (geophile) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/qp/storeadapter/indexcursor/IndexCursorSpatial_InBox.java]: I'll fix this too.

File: fdb-sql-layer-core/src/main/java/com/foundationdb/server/spatial/Spatial.java (hunk 3 above)
Comment (geophile) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/server/spatial/Spatial.java]: I intend to use this to turn down geophile log noise (punchlist item 11).

**Comment by geophile** [comment]
This should include the 1668 changes.

**Comment by geophile** [comment]
Sorry, I thought I responded to the question about punchlist item 1 already.

Yes, those changes are also present, and I updated the pull request title to indicate this.

**Comment by nathanlws** [comment]
Thanks for the tweaks. Looks as discussed.

Approve

**Pull request merged**